src/zenml/config/settings_resolver.py

```diff
@@ -122,7 +122,7 @@ def _convert_settings(self, target_class: Type["T"]) -> "T":
         Returns:
             The converted settings.
         """
-        settings_dict = self._settings.dict()
+        settings_dict = dict(self._settings)
```

Comment:
Just in case, would it be cleaner, if we use `model_dump` everywhere for such tasks?

Reply:
Unfortunately, `model_dump` and `dict` do not return the same thing:

```python 
from pydantic import BaseModel


class One(BaseModel):
    a: int = 3


class Two(BaseModel):
    b: str = "3"
    c: One = One()


print(Two().model_dump())  # -> {'b': '3', 'c': {'a': 3}}
print(dict(Two()))  # -> {'b': '3', 'c': One(a=3)}
```

Based on the logic, I had to change the code accordingly.